### ISSUE TITLE
write-api admin Page not working Not able to generate  bearer token

### ISSUE THREAD

**Issue by drlukeangel** [NodeBB/nodebb-plugin-write-api#76]
Generate your uid an API token in the ACP(?admin control panel?) page.
Going the to url directly admin/plugins/write-api throughs a 404
![image](https://user-images.githubusercontent.com/22433329/31258888-61daf994-a9f7-11e7-9b1e-4064a2105cdc.png)
plugin is installed
![image](https://user-images.githubusercontent.com/22433329/31258945-d1b68b20-a9f7-11e7-9158-7a73090f8480.png)
Not showing in drop down not sure it would though
![image](https://user-images.githubusercontent.com/22433329/31258929-b665ddee-a9f7-11e7-9794-76f152365828.png)
event shows activation 
![image](https://user-images.githubusercontent.com/22433329/31259056-b41a5e10-a9f8-11e7-8d89-ca81ab385346.png)
did a restart as requested as well
Seems to of installed correctly
![image](https://user-images.githubusercontent.com/22433329/31258989-320dbcc8-a9f8-11e7-84a9-297429ddce2e.png)
Feels like a bad route
running 
![image](https://user-images.githubusercontent.com/22433329/31259092-f32469ca-a9f8-11e7-9919-b1e20f28df48.png)

**Comment by drlukeangel** [comment]
so debugging this a bit it seems that 
in themes.js
I am currently using the "nodebb-theme-material"
when you iterate through the themes to get the themeObj it can't find it so sets themeObj and null 
	var themeObj = data.themesData.filter(function (themeObj) {
				return themeObj.id === themeId;
			})[0];
causing the if statement to through the 
				return callback(new Error('[[error:theme-not-found]]'));
and making then making the new build break. 
so commenting that out makes initial build work

**Comment by julianlam** [comment]
@pichalite can you reproduce?

**Comment by drlukeangel** [comment]
@julianlam 
*TLDR -  Starting with a fresh install the write_api worked but using a existing install with material theme eventually corrupted the entire mongodb. 
-----
From a end user admin perspective I followed the instructions it asked for. Installed it via the plugin and then restarted the instance. If restarting via the admin portal does a full build I wouldn't know. 
From a dev perspective. I stopped the server verified the plugin installed in the node_modules and then did a npm install as stated in the plugin readme.md from the plugin folder to super verify that write_api was installed and I had all the libraries. When I restarted the server again this is with the theme bug commented out it mongo db became corrupted and it would not restart. So I took that as a sign and just trashed the mongo server I am using docker so this was not a big deal started a new one and then did a fresh app db install same code from before just the db init was new. This time I did not install/activatel the material theme, and then activated(no install as it was already in the folder from last install) the write plugin and it worked fine.

**Comment by pichalite** [comment]
@julianlam @drlukeangel Can't reproduce. Nothing related to theme.

I have the material theme installed and active and was able to install/activate the write-api plugin without any issues. 

![image](https://user-images.githubusercontent.com/10014774/31288537-2b760ca8-aa7a-11e7-8ad1-b70041c17afb.png)

![image](https://user-images.githubusercontent.com/10014774/31288563-483c8a2e-aa7a-11e7-825e-cf27ccfd0879.png)

The plugin itself not showing up in the menu indicates the build step was either not triggered or failed.

@drlukeangel Just for information, what version of Nodejs and npm are you using?

**Comment by drlukeangel** [comment]
v6.11.2 - feel free to close this maybe it glitched somewhere i wasn't aware of. I as well got it to work but only after dropping the datastore. I haven't applied material to the new instance but ill comment on this if does glitch

**Comment by julianlam** [comment]
Thank you all for your attention
